Modules/Core/Transform/include/itkComposeScaleSkewVersor3DTransform.hxx

```diff
@@ -0,0 +1,372 @@
+  M(0, 2) -= (ortho0 * M(0, 0) + ortho1 * M(0, 1));
+  M(1, 2) -= (ortho0 * M(1, 0) + ortho1 * M(1, 1));
+  M(2, 2) -= (ortho0 * M(2, 0) + ortho1 * M(2, 1));
+  scaleV[0] = M(0, 2);
+  scaleV[1] = M(1, 2);
+  scaleV[2] = M(2, 2);
+  m_Scale[2] = scaleV.GetVnlVector().magnitude();
+  M(0, 2) /= m_Scale[2];
+  M(1, 2) /= m_Scale[2];
+  M(2, 2) /= m_Scale[2];
+  m_Skew[1] = ortho0 / m_Scale[0];
+  m_Skew[2] = ortho1 / m_Scale[1];
+  if (vnl_determinant(M.GetVnlMatrix()) < 0)
+  {
+    m_Scale[0] *= -1;
+    M(0, 0) *= -1;
+    M(1, 0) *= -1;
+    M(2, 0) *= -1;
+  }
+
+  VersorType v;
+  v.Set(M);
+  this->SetVarVersor(v);
+}
+
+// Print self
+template <typename TParametersValueType>
+void
+ComposeScaleSkewVersor3DTransform<TParametersValueType>::PrintSelf(std::ostream & os, Indent indent) const
+{
+  Superclass::PrintSelf(os, indent);
+
+  os << indent << "Scale:       " << m_Scale << std::endl;
+  os << indent << "Skew:        " << m_Skew << std::endl;
+}
+
+template <typename TParametersValueType>
+void
+ComposeScaleSkewVersor3DTransform<TParametersValueType>::ComputeJacobianWithRespectToParameters(
+  const InputPointType & p,
+  JacobianType &         jacobian) const
+{
+  using ValueType = typename VersorType::ValueType;
+
+  // compute derivatives with respect to rotation
+  const ValueType vx = this->GetVersor().GetX();
+  const ValueType vy = this->GetVersor().GetY();
+  const ValueType vz = this->GetVersor().GetZ();
+  const ValueType vw = this->GetVersor().GetW();
+
+  jacobian.SetSize(3, this->GetNumberOfLocalParameters());
+  jacobian.Fill(0.0);
+
+  const double px = p[0] - this->GetCenter()[0];
+  const double py = p[1] - this->GetCenter()[1];
+  const double pz = p[2] - this->GetCenter()[2];
+
+  const double vxx = vx * vx;
+  const double vyy = vy * vy;
+  const double vzz = vz * vz;
+  const double vww = vw * vw;
+
+  const double vxy = vx * vy;
+  const double vxz = vx * vz;
+  const double vxw = vx * vw;
+
+  const double vyz = vy * vz;
+  const double vyw = vy * vw;
+
+  const double vzw = vz * vw;
+
+  // compute Jacobian with respect to quaternion parameters
+  jacobian[0][0] = 2.0 * ((vyw + vxz) * py + (vzw - vxy) * pz) / vw;
+  jacobian[1][0] = 2.0 * ((vyw - vxz) * px - 2 * vxw * py + (vxx - vww) * pz) / vw;
+  jacobian[2][0] = 2.0 * ((vzw + vxy) * px + (vww - vxx) * py - 2 * vxw * pz) / vw;
+
+  jacobian[0][1] = 2.0 * (-2 * vyw * px + (vxw + vyz) * py + (vww - vyy) * pz) / vw;
+  jacobian[1][1] = 2.0 * ((vxw - vyz) * px + (vzw + vxy) * pz) / vw;
+  jacobian[2][1] = 2.0 * ((vyy - vww) * px + (vzw - vxy) * py - 2 * vyw * pz) / vw;
+
+  jacobian[0][2] = 2.0 * (-2 * vzw * px + (vzz - vww) * py + (vxw - vyz) * pz) / vw;
+  jacobian[1][2] = 2.0 * ((vww - vzz) * px - 2 * vzw * py + (vyw + vxz) * pz) / vw;
+  jacobian[2][2] = 2.0 * ((vxw + vyz) * px + (vyw - vxz) * py) / vw;
+
+  jacobian[0][3] = 1.0;
+  jacobian[1][4] = 1.0;
+  jacobian[2][5] = 1.0;
+
+  jacobian[0][6] = px;
```

Comment:
The scale dependent Jacobian values an incorrect as well.

Maybe this whole implementation should be removed and only leave the exception?

Reply:
Removed and appropriate comments added.